engine/platform/sdl2/vid_sdl2.c

```diff
@@ -1164,6 +1165,54 @@ qboolean VID_SetMode( void )
 	return true;
 }
 
+qboolean R_GetWindowHandle( void **handle, int type )
+{
+	SDL_SysWMinfo wmInfo;
+	SDL_VERSION( &wmInfo.version );
+
+	if ( SDL_GetWindowWMInfo( host.hWnd, &wmInfo ) != SDL_TRUE )
+	{
+		return FALSE;
+	}
+
+	switch( type )
```

Comment:
Again, you can not be sure that requested window handle type matches whatever SDL used to create the window.

I think the correct logic is like this:

```
switch( type )
{
case REF_WINDOW_TYPE_WIN32:
#ifdef SDL_VIDEO_DRIVER_WINDOWS
	if( wmInfo.info.subsystem == SDL_SYSWM_WINDOWS )
	{
		*handle = (void*)wmInfo.info.win.window;
		return true;
	}
	else
		return false;
#else
	return false;
#endif
```

Also `case` doesn't indent, it stays on the same level as `switch`.

Reply:
Also, on some platforms like Linux it might make sense to query window type FIRST and only then fetch handle.

Imagine the situation when we don't know if X11 or Wayland is used, though renderer supports both of them.

Returning a `struct` with handle _and_ window type was a good idea (I don't know why you decided to go with the void-pointer though), this way renderer might not ask for specific window type.